src/btree/bt_split.c

```diff
@@ -986,6 +985,18 @@ __split_parent(WT_SESSION_IMPL *session, WT_REF *ref, WT_REF **ref_new,
 					    session, 0, ikey, size));
 					WT_MEMSIZE_ADD(parent_decr, size);
 				}
+				/*
+				 * The page_del structure can be freed
+				 * immediately: it is only read when the ref
+				 * state is WT_REF_DELETED.  The size of the
+				 * structures wasn't added to the parent: don't
+				 * decrement.
+				 */
+				if (next_ref->page_del != NULL) {
+					__wt_free(session,
+					    next_ref->page_del->update_list);
+					__wt_free(session, next_ref->page_del);
+				}
```

Comment:
Got it -- sorry, I missed that.

Still... there's no reason to keep the memory around, is there?

Either way, I'm good to go.